pkg/storage/etcd3/watcher.go

```diff
@@ -109,6 +107,10 @@ func (wc *watchChan) run() {
 
 	select {
 	case err := <-wc.errChan:
+		if err == context.Canceled {
+			wc.cancel() // just in case
```

Comment:
This is sort of last defense line -- what if we get context.Canceled but `wc.ctx` isn't canceled.
We want to make sure all goroutines in etcd3 watcher to stop regardlessly.